### PR TITLE
Add support for general info

### PR DESCRIPTION
**Pull request by ghostbuster91**
This refers to #11 

I am not sure about this duplication of classes (`Info`, `Contact`, `Licence`) since their purpose is only to hold information relevant for openApi. So they could live in openApi module and be provided as we convert endpoint to openApi structures.

On the other hand there is e.g. a root server structure which may affects the behavior of endpoints, so it will have to have a representation in tapir module.

I wonder whether we should keep parts of openApi structers only in relevant modules or to have consistent api with duplicates.

An example:
1. with duplicates
```
val endpoint: Endpoint = ....
val api = tapir.Api(info = ..., endpoints= ..., server = ...)  <- all those things live in tapir module
val openApi = api.toOpenAPI
val http4sRoute = endpoint.toHttp4s(api)
```
2. 
```
val endpoint: Endpoint = ....
val server = tapir.Server
val info = openapi.Info
val openApi = endpoint.toOpenapi(server, info)
val http4sRoute = endpoint.toHttp4s(server)
```

Update:
Since there can be multiple servers specified we always need to explicitly provide a particular instance.

### CODE REVIEW

File: core/src/main/scala/tapir/Api.scala [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  )
+  case class Contact(name: Option[String], email: Option[Email], url: Option[URL])
+  case class License(name: String, url: Option[URL])
+
+  def apply(title: String, version: String): Api = {
+    new Api(info = Info(title, version))
+  }
+
+  type Email = String
+  type URL = String
```
Comment (ghostbuster91) [inline core/src/main/scala/tapir/Api.scala]: I need here and in some other places and URI class with parse. Should I just copy Uri form sttp code for now or use java.net.URI?

Comment (adamw) [reply core/src/main/scala/tapir/Api.scala]: I think `java.net.URI` should be sufficient. The main benefit of the sttp one is building URIs through interpolation, which we don't really need here

File: core/src/main/scala/tapir/Api.scala [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package tapir
+import tapir.Api.Info
+
+case class Api(info: Info)
```
Comment (adamw) [inline core/src/main/scala/tapir/Api.scala]: I'm not sure if this should be in the core module. It's only ever going to be used by docs, no?

Also, why `case class Api(Info)`, and not just `case class ApiInfo(...)` ?

**Comment by adamw** [comment]
I think I would avoid duplicates as long as possible - that is, once there's a really use-case for `Api`-like classes outside of documentation generation (be it openapi or not), let's leave them in the docs modules. 

That might be also intersecting with #31: to conveniently define the openapi metadata structures, you might need some builder methods/alternate constructors etc., which would directly yield e.g. `OpenAPI`, `Info` etc.

File: docs/openapi-docs/src/main/scala/tapir/docs/openapi/OpenAPIDocs.scala [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
-    def toOpenAPI(title: String, version: String)(implicit options: OpenAPIDocsOptions): OpenAPI =
-      EndpointToOpenAPIDocs.toOpenAPI(title, version, Seq(e), options)
+    def toOpenAPI(info: Info)(implicit options: OpenAPIDocsOptions): OpenAPI =
+      EndpointToOpenAPIDocs.toOpenAPI(info, Seq(e), options)
   }
 
   implicit class RichOpenAPIEndpoints(es: Iterable[Endpoint[_, _, _, _]]) {
-    def toOpenAPI(title: String, version: String)(implicit options: OpenAPIDocsOptions): OpenAPI =
-      EndpointToOpenAPIDocs.toOpenAPI(title, version, es, options)
+    def toOpenAPI(info: Info)(implicit options: OpenAPIDocsOptions): OpenAPI =
```
Comment (adamw) [inline docs/openapi-docs/src/main/scala/tapir/docs/openapi/OpenAPIDocs.scala]: a może zostawić oba warianty: jeden z `title: String, version: String` i drugi z `info: Info`? Ten drugi by dawał pełną elastycznośc, pierwszy by zapewniał "szybką ścieżkę"?

**Pull request merged**